app/views/admin/services/generate_magic_links.html.erb

```diff
@@ -0,0 +1,33 @@
+
+<h1>Magic links</h1>
+
+<% @service_data.each do |service, dates| %>
+<h2><%= service.name %></h2>
+<table class="index_table index"">
+  <thead>
+    <tr>
+      <th>Month</th>
+      <th>Year</th>
+      <th>Owner</th>
+      <th>Link</th>
+    </tr>
+  </thead>
+  <tbody>
+    <% dates.each do |date, token| %>
+    <tr>
+      <td><%= date.year %></td>
+      <td><%= date.strftime("%B") %></td>
+      <td><%= service.owner.email if service.owner %></td>
+      <td>
+        <%= absolute_url(url_for publish_service_metrics_path(
```

Comment:
I _think_ it should be possible to just use `publish_service_metrics_url` here. That should give you the absolute URL.